src/battle_anim_mon_movement.c

```diff
@@ -554,15 +550,9 @@ static void SlideMonToOriginalPos_Step(struct Sprite *sprite)
 // arg 4: duration
 static void SlideMonToOffset(struct Sprite *sprite)
 {
-    u8 battler;
-    u8 monSpriteId;
-    if (!gBattleAnimArgs[0])
-        battler = gBattleAnimAttacker;
-    else
-        battler = gBattleAnimTarget;
+    u8 monSpriteId = GetAnimBattlerSpriteId(gBattleAnimArgs[0]);
 
-    monSpriteId = gBattlerSpriteIds[battler];
-    if (GetBattlerSide(battler) != B_SIDE_PLAYER)
+    if (GetBattlerSide(gBattleAnimArgs[0]) != B_SIDE_PLAYER)
```

Comment:
`gBattleAnimArgs[0]` does not contain the battler Id, just the relative anim Id. It needs to be passed through GetAnimBattlerId first to get the battler.